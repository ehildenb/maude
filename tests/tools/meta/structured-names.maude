set show timing off .

load ../../../contrib/tools/meta/structured-names.maude

fmod TEST-STRUCTURED-NAME is
   protecting STRUCTURED-NAME .

    ops X Y Z F : -> Sort .
    -----------------------
    eq X = var<Sort>('X) .
    eq Y = var<Sort>('Y) .
    eq Z = var<Sort>('Z) .
    eq F = var<Sort>('F) .

    op sortName : Nat -> Sort .
    ---------------------------
    eq sortName(0) = X .
endfm

--- Module STRUCTURED-NAMES
--- -----------------------

reduce prime(X) .
reduce prime(3, X) .
reduce prime(3, F{X}) .

reduce prime(X, F{X}) .
reduce prime(2, X, F{X}) .

reduce prime(F{X}, F{F{X}}) .
reduce prime(F{X}, F{(F{X}) (F{X})}) .
