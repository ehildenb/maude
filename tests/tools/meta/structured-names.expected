Warning: sort declarations for operator resolveNames failed preregularity check
    on 6 out of 38 sort tuples. First such tuple is (Type).
==========================================
reduce in TEST-STRUCTURED-NAME : prime(X) .
rewrites: 33
result Sort: X':Sort
==========================================
reduce in TEST-STRUCTURED-NAME : prime(3, X) .
rewrites: 36
result Sort: X''':Sort
==========================================
reduce in TEST-STRUCTURED-NAME : prime(3, F{X}) .
rewrites: 71
result Sort: F''':Sort{X''':Sort}
==========================================
reduce in TEST-STRUCTURED-NAME : prime(3, 'F{X}) .
rewrites: 39
result Sort: 'F{X''':Sort}
==========================================
reduce in TEST-STRUCTURED-NAME : prime(X, X) .
rewrites: 39
result Sort: X':Sort
==========================================
reduce in TEST-STRUCTURED-NAME : prime(X, Y) .
rewrites: 45
result Sort: Y:Sort
==========================================
reduce in TEST-STRUCTURED-NAME : prime(X, F{X}) .
rewrites: 48
result Sort: F:Sort{X':Sort}
==========================================
reduce in TEST-STRUCTURED-NAME : prime(X, 'F{X}) .
rewrites: 42
result Sort: 'F{X':Sort}
==========================================
reduce in TEST-STRUCTURED-NAME : prime(2, X, F{X}) .
rewrites: 49
result Sort: F:Sort{X'':Sort}
==========================================
reduce in TEST-STRUCTURED-NAME : prime(Y, F{X}) .
rewrites: 54
result Sort: F:Sort{X:Sort}
==========================================
reduce in TEST-STRUCTURED-NAME : prime(X, F{F{X}}) .
rewrites: 51
result Sort: F:Sort{F:Sort{X':Sort}}
==========================================
reduce in TEST-STRUCTURED-NAME : prime(X, F{(F{X}) (F{X})}) .
rewrites: 57
result Sort: F:Sort{(F:Sort{X':Sort}) (F:Sort{X':Sort})}
==========================================
reduce in TEST-STRUCTURED-NAME : prime(F, F{(F{X}) (F{X})}) .
rewrites: 57
result Sort: F':Sort{(F':Sort{X:Sort}) (F':Sort{X:Sort})}
Bye.
